### PR TITLE
#975 fix Move brackets format function out of StringFrom to keep general usage on StringFrom

### PR DESCRIPTION
**Pull request by cuitoldfish**

### CODE REVIEW

File: src/CppUTest/SimpleString.cpp [hunk 1]
```diff
@@ -492,7 +513,7 @@ SimpleString StringFrom(cpputest_longlong value)
 
 SimpleString StringFrom(cpputest_ulonglong value)
 {
-    return StringFromFormat("%llu (0x%llx)", value, value);
+    return StringFromFormat("%llu", value, value);
```
Comment (basvodde) [inline src/CppUTest/SimpleString.cpp]: Fails here as you pass value twice to the StringFromFormat :)

Comment (cuitoldfish) [inline src/CppUTest/SimpleString.cpp]: will correct soon :)

**Comment by arstrube** [comment]
@basvodde do we really need a bracketed hex string? I would prefer to get rid of it. Or, if we want bracketed hex strings in test failures, then add the brackets there (but personally, I feel they are redundant). I feel the few places where it was printed with brackets (plus without) were more of a slip than an intended feature.

@cuitoldfish if we get rid of the brackets, this PR will become a whole lot simpler.

**Comment by cuitoldfish** [comment]
yes, but I'm hesitate if some modification will change what people has already currently accepted, just like this brackets format hex string.

I'm also curious why currently there is no brackets hex format for signed type(only for unsigned type)?

**Comment by arstrube** [comment]
It's because the brackets were an inconsistancy to start with..... I feel they are not needed for anything, a hex number can easily be parsed looking for the 0x....

I was in fact wondering why we accepted tests like that, e.g. `<1 (0x1) 0x1>` when it was plain to see that all other tests were checking for `<1 0x1>`.

**Comment by basvodde** [comment]
I'd somehow prefer the 1 (0x01) format for all. The brackets make it clearer that it is a representation of the same format.

**Comment by arstrube** [comment]
@basvodde that would work, too.... would be still need the extra set of methods though? I would prefer to see the adding of brackets outside of SimpleString. I think SimpleString should do the simple thing :).

**Comment by basvodde** [comment]
We passed the simple part long time ago :)

I think its ok in SimpleString as it would be at least in one place. Otherwise the same format would be in multiple places, which is how we ended up with the problem in the first place.

File: tests/TestUTestMacro.cpp [hunk 2]
```diff
@@ -77,8 +77,8 @@ static void _UNSIGNED_LONGS_EQUALTestMethod()
 TEST(UnitTestMacros, TestUNSIGNED_LONGS_EQUAL)
 {
     fixture.runTestWithMethod(_UNSIGNED_LONGS_EQUALTestMethod);
-    CHECK_TEST_FAILS_PROPER_WITH_TEXT("expected <1 (0x1) 0x1>");
-    CHECK_TEST_FAILS_PROPER_WITH_TEXT("but was  <0 (0x0) 0x0>");
+    CHECK_TEST_FAILS_PROPER_WITH_TEXT("expected <1 0x1>");
+    CHECK_TEST_FAILS_PROPER_WITH_TEXT("but was  <0 0x0>");
```
Comment (arstrube) [inline tests/TestUTestMacro.cpp]: @basvodde if we do the brackets thing, these need to be bracketed, too

Comment (arstrube) [inline tests/TestUTestMacro.cpp]: and I would have to make the same change to SIGNED_BYTES_EQAUL & friends.

Comment (basvodde) [inline tests/TestUTestMacro.cpp]: Correct

**Comment by arstrube** [comment]
@basvodde okay you convinced me there ;-)

File: tests/TestUTestMacro.cpp (hunk 2 above)
Comment (arstrube) [inline tests/TestUTestMacro.cpp]: This one will then need to be merged before my signed bytes one.

**Comment by cuitoldfish** [comment]
how about cpputest_longlong type,  I didn't create a brackets format function for it because I didn't see where this type is printed in a string.

**Comment by cuitoldfish** [comment]
So for 
```.cpp
CHECK_TEST_FAILS_PROPER_WITH_TEXT("but was  <0 (0x0) 0x0>");
```
will be changed to 
```.cpp
CHECK_TEST_FAILS_PROPER_WITH_TEXT("but was  <0 (0x0)>");
```
?

and for signed type, take -1 as an example,  it's hex string will be 0xFFFFFFFF, so in the above case, that would be
```.cpp
CHECK_TEST_FAILS_PROPER_WITH_TEXT("but was  <-1 (0xFFFFFFFF)>");
```
still looks good?

**Comment by arstrube** [comment]
yes, looks good to me :)

**Comment by cuitoldfish** [comment]
if the brackets hex format string is part of all numeric type string, then I prefer to add this just inside StringFrom,

**Comment by arstrube** [comment]
yes I think that's the idea, just the methods you already added.

**Comment by arstrube** [comment]
umm, no, not in StringFrom(). That is used in many other places, and we don't want brackets there!

**Comment by arstrube** [comment]
maybe I'm wrong. Try what happens if you put it into StringFrom, if any tests fail, etc. I am not a great fan of the brackets, but @basvodde likes them. Perhaps it is okay in StringFrom, although I would have preferred a 'normal' string format. Also, should now suddenly pointers be in brakcets, when it is the main displayed value? I think not....

**Comment by arstrube** [comment]
I think your PR is fine with regard to SimpleString. Just need to use brackets consistently for all failures and in all tests, or? (persoanlly I like it better as is, though)...

**Comment by basvodde** [comment]
Not in the StringFrom, that would be a mistake :)

**Pull request closed without merging**